### PR TITLE
Created the functionality to split the columns for issue #85

### PR DESCRIPTION
**Pull request by puneetsharma04**
@SemyonSinchenko : Could you please review the code changes and give the comments.

### CODE REVIEW

File: quinn/split_columns.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    Splits the given column based on the delimiter and creates new columns with the split values.
+    :param df: The input DataFrame
+    :type df: pyspark.sql.DataFrame
+    :param col_name: The name of the column to split
+    :type col_name: str
+    :param delimiter: The delimiter to split the column on
+    :type delimiter: str
+    :param new_col_names: A list of two strings for the new column names
+    :type new_col_names: (List[str])
+    :param mode: The split mode. Can be "strict" or "permissive". Default is "strict"
```
Comment (SemyonSinchenko) [inline quinn/split_columns.py:18]: Looks like default now is permissive. But better to have strict here.

Comment (puneetsharma04) [reply quinn/split_columns.py:18]: Made changes to code as suggested.

**Comment by MrPowers** [comment]
@SemyonSinchenko - I'm fine having this functionality in a separate file.   It's actually nice having a separate file.  There should be like 10 tests, so having the separation is nice.

@puneetsharma04 - can you please add an assertion to the test?  Can you also add different tests to cover all the edge cases?  Thank you!

**Comment by MrPowers** [comment]
xref: https://github.com/MrPowers/quinn/issues/85

**Comment by puneetsharma04** [comment]
@MrPowers  @SemyonSinchenko : Thanks for the review comments , i will make the code changes to include test case and commit the changes.

**Comment by puneetsharma04** [comment]
@SemyonSinchenko & @MrPowers : Please review the test cases and share you review & comments.

File: quinn/split_columns.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    split_col_expr = split(col(col_name), delimiter)
+
+    if mode == "strict":
+        df = df.select("*", split_col_expr.getItem(0).alias(new_col_names[0]),
+                       split_col_expr.getItem(1).alias(new_col_names[1])) \
+            .drop(col_name)
+    elif mode == "permissive":
+        df = df.select("*", split_col_expr.getItem(0).alias(new_col_names[0]),
+                       when(length(split_col_expr.getItem(1)) > 1, split_col_expr.getItem(1))
```
Comment (SemyonSinchenko) [inline quinn/split_columns.py:40]: May you please explin this a little? What will happen when mode is permissive but there are only two parts and `length(split_col_expr.getItem(1) == 1`?

File: quinn/split_columns.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                       .alias(new_col_names[1])) \
+            .filter(col(new_col_names[1]).isNotNull()) \
```
Comment (SemyonSinchenko) [inline quinn/split_columns.py:42]: Please may you explain a little why should we filter rows?

**Review by SemyonSinchenko — CHANGES_REQUESTED** [review]

File: quinn/split_columns.py (hunk 2 above)
Comment (puneetsharma04) [reply quinn/split_columns.py:40]: @SemyonSinchenko : Thanks for the review comment, explanation is below, Let me know in case if expectation is different from this:
This line of code uses the when function to check if the length of the second part of the split result is greater than 1. If it is, the second new column will be created with the second part of the split result. If not, the second new column will be created with null.
So, if mode is "permissive" and there are only two parts after the split but the second part has a length of 1, then the function will create the new columns accordingly but the second new column will be null. The resulting DataFrame will contain a row where the first new column contains the first part of the split result and the second new column is null.

File: quinn/split_columns.py (hunk 3 above)
Comment (puneetsharma04) [reply quinn/split_columns.py:42]: The reason for filtering rows in this line of code is to remove any rows where the second part of the split result is null or empty. In "permissive" mode, when the split result has only one part, the function creates a new column with null as its value. If there are other rows in the DataFrame where the second part of the split result is null or empty, then these rows will also have a null value in the new column.

Therefore, after creating the new columns using the select method, the function applies a filter to remove any rows where the second new column is null or empty using the isNotNull method.

**Comment by puneetsharma04** [comment]
@SemyonSinchenko : Thanks for sharing the review comments.Let me refactor the code and share again.
Could you please check the below given scenario and share the comments:
mode: strict
intitial value: "val1_val2"
new_col_names: ["col1", "col2"]
RESULT: for some records if val2 is null or blank, then how would you like it to be handled ?

**Comment by SemyonSinchenko** [comment]
> @SemyonSinchenko : Thanks for sharing the review comments.Let me refactor the code and share again.
> Could you please check the below given scenario and share the comments:
> mode: strict
> intitial value: "val1_val2"
> new_col_names: ["col1", "col2"]
> RESULT: for some records if val2 is null or blank, then how would you like it to be handled ?


In this case we should raise an Exception because mode is strict. Strict mode means that you should check the length of result of the split.

**Comment by puneetsharma04** [comment]
@SemyonSinchenko : Could you please review the code changes and share your comments if it matches the expectations.

**Comment by SemyonSinchenko** [comment]
> @SemyonSinchenko : Could you please review the code changes and share your comments if it matches the expectations.

A lot of changes in the last commit.. I do it in a weekend. Sorry for such a delay: a lot of work :(
